core/src/main/java/org/apache/calcite/rel/rules/ProjectToWindowRule.java

```diff
@@ -240,6 +246,11 @@ public ProjectToLogicalProjectAndWindowRule(
             return false;
           }
 
+          // TODO: Is this correct?
```

Comment:
Seems like it? I don't fully understand how this rule works.

Reply:
Same reasoning as above.